### ISSUE TITLE
CheckCounter for processes using over 100% CPU

### ISSUE THREAD

**Issue by Klang0340** [mickem/nscp#264]
Good afternoon all,

    Looking for assistance to identify the correct way to use CheckCounter (NSClient 5.x) for a process using over 100% CPU. My example is sqlserver.exe uses 1200% if it has 12 Cores and is maxed out.

In the below thread, the 'nocap100' functionality was added but I couldn't get this to work when I tested. Most likly I was using the wrong syntax BUT then I noticed in the help text ShowAll=long may be the new option for this.
http://docs.nsclient.org/0.4.3/whatsnew/changelog.html

I've been unable to get checkCounter to report more than 100% CPU utilization for processes even though the performance counter is currently reporting 1000% and higher. Below are the two best examples I have used to test.

Example 1 using 'nocap' based on change notes
/opt/plugins/check_nrpe -u -t 15 -H <IP ADDRESS> -c CheckCounter -a 'Counter=\Process(sqlservr)\% Processor Time' **ShowAll=nocap100**
OK: \Process(sqlservr)\% Processor Time = 100|'\Process(sqlservr)\% Processor Time'=100;0;0

Example 2 using 'ShowAll=long' based on help text for checkCounter
/opt/plugins/check_nrpe -u -t 15 -H <IP ADDRESS> -c CheckCounter -a 'Counter=\Process(sqlservr)\% Processor Time' **ShowAll=long**
OK: \Process(sqlservr)\% Processor Time = 100|'\Process(sqlservr)\% Processor Time'=100;0;0

Basically, in the two examples above the % Processor Time should be well over 100%.

**Comment by mickem** [comment]
Haven tried with sql server but adding the flags is done like so:

```
check_pdh "counter=\\Process(sqlservr)\\% Processor Time" show-all flags=nocap100
```

**Comment by mickem** [comment]
(pr from nrpe)
```
/opt/plugins/check_nrpe -u -t 15 -H -c check_pdh -a "counter=\\Process(sqlservr)\\% Processor Time" show-all flags=nocap100
```

**Comment by Klang0340** [comment]
Michael,

                This worked great, can you explain the use case for check_pdh? I haven’t seen this before but I’m reading the documentation now on http://docs.nsclient.org/samples/index.html#check-pdh.

Attempt #1 – At first, I thought I would use the syntax your provided and this didn’t work. I then realized you are using ‘check_pdh’ versus ‘CheckCounter’.
[monitor@opmaster1-corp-p1 /opt/plugins/custom/wrapper]$ /opt/plugins/check_nrpe -u -t 15 -H <IP_Address> -c CheckCounter -a 'Counter=\Process(sqlservr)\% Processor Time' show-all flags=nocap100
OK: \Process(sqlservr)\% Processor Time = 100|'\Process(sqlservr)\% Processor Time'=100;0;0

Attempt #2 – Using your exact syntax I got what needed, below you can see sql server was using well over 100% cpu which is what I needed. Thank you!
/opt/plugins/check_nrpe -u -t 15 -H <IP_Address> -c check_pdh -a 'counter=\Process(sqlservr)\% Processor Time' show-all flags=nocap100
OK: \Process(sqlservr)\% Processor Time = 830|'\Process(sqlservr)\% Processor Time_value'=830;0;0

**Comment by mickem** [comment]
check_pdh replaces CheckCounter.
From 0.4.3 and beyon all "old" commands are wrapper for "new commands" with _ in the name.
So when you run checkCounter it will rewrite the query to a check_pdh command.

The "new" commands are more consistent and have a simpler yet more powerful syntax.
With the new syntax there is no ordering of variables and all commands have the same set of options and they all work the same.

**Comment by Klang0340** [comment]
Understood,

                Thank you for this information!
